src/crimson/os/seastore/transaction.h

```diff
@@ -90,6 +89,25 @@ class Transaction {
     write_set.insert(*ref);
   }
 
+  void replace_placeholder(CachedExtent& to, CachedExtent& from) {
+    ceph_assert(!is_weak());
+
+    assert(from.get_type() == extent_types_t::RETIRED_PLACEHOLDER);
+    assert(to.get_type() != extent_types_t::RETIRED_PLACEHOLDER);
+    assert(to.get_type() != extent_types_t::ROOT);
+    assert(to.get_paddr() == from.get_paddr());
+
+    auto iter = read_set.find(from.get_paddr());
+    assert(iter != read_set.end());
+    assert(iter->ref.get() == &from);
+    read_set.erase(iter);
+    auto erased = retired_set.erase(&from);
+    assert(erased);
+
+    add_to_read_set(&to);
+    retired_set.insert(&to);
```

Comment:
if the placeholder extent and the new one extent share the same paddr, why not just 
```c++
    {
      auto where = read_set.find(from.get_paddr());
      assert(where != read_set.end());
      assert(where->ref.get() == &from);
      where = read_set.erase(where);
      read_set.emplace_hint(where, this, &to);
    }
    {
      auto where = retired_set.find(from.get_paddr());
      assert(where != retired_set.end());
      where = retired_set.erase(where);
      retired_set.emplace_hint(where, &from);
    }
```

for saving the look up.

Reply:
done.